### PR TITLE
draft of the blog post

### PR DESCRIPTION
**Pull request by rjmk**
Hey, here is the blog requested at https://github.com/rub1e/beam-me-up-hapi/issues/10

Let me know any edits, suggestions, improvements, expansions, corrections or alterations you have.

### CODE REVIEW

File: file-upload-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# A tutorial on doing file uploads in Hapi
+
+### Why file uploads
+
+Every website needs content, but not every website needs to create its own content. If you want your users to be able to share images, music or video with their friends and with the world, you have to crack file uploads.
```
Comment (iteles) [inline file-upload-tutorial.md]: Would be nice to have a more specific example of a use case here :)

File: file-upload-tutorial.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+### What we made
+
+We have built a simple website using the Hapi framework for Node.js. It provides a form for the users that allows them to submit files to the server. The server then saves that in a 'pix' directory (though the files do not have to be pictures!).
```
Comment (iteles) [inline file-upload-tutorial.md]: Suggest adding links to the hapi and nodeJS sites

File: file-upload-tutorial.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+To aid the rest of our FAC cohort, we have developed a repo that provides a tutorial on building the site.
```
Comment (iteles) [inline file-upload-tutorial.md]: 'Unexplained' FAC abbreviation for external audience. Also, I would add the word 'github' infront of repo

File: file-upload-tutorial.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+### On Modules
+
+Certain modules can help in the parsing of complicated input to the server -- in this case, the multipart form is the complex data. Some examples make use of the 'multiparty' node module to help with parsing it. We have chosen to abstain from doing so. There are three reasons why:
```
Comment (iteles) [inline file-upload-tutorial.md]: Suggest adding a clarification of 'multipart form' or link to an explanation (usually a good answer to a stackoverflow question or article online) - seeing as this is for a wide audience that includes beginners

File: file-upload-tutorial.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  * If you write all the code yourself, it's simpler to understand and debug in simple cases. There's a fairly fixed overhead in understanding the syntax and functionality of an external module. This is often a very good tradeoff, particularly if it's a module that you will be using frequently. In the case of a toy example, it would probably have been more trouble than it's worth.
+
+  * Thirdly, Hapi has some parsing ability of its own by default. We're trying to understand Hapi and what it provides, so there's not much point overwriting that capability with an extra module. If we had followed the examples we found online, we may never have known about Hapi's native parsing abilities.
+
+### The tutorial and how to use it
+
+[The repo has different stages of the website on different branches](http://www.github.com/foundersandcoders/beam-me-up-hapi). After cloning the repo on your machine, start on the 0th branch by `git checkout 0` on the command line. You can then step through the readme, which provides you with some guidance on how to reach the next stage. If you get stuck or frustrated, you can always checkout to the next branch to see how we did it. Happy coding!
+
+FAC 5
```
Comment (iteles) [inline file-upload-tutorial.md]: @sofer This raises a good question of attribution for blog posts - how is it/should it be done?! :)

File: file-upload-tutorial.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
+# A tutorial on doing file uploads in Hapi
+
```
Comment (iteles) [inline file-upload-tutorial.md]: Personally, I like to know on the first line what is happening in the rest of the blog post. This is a very personal opinion but I would suggest you consider adding an introductory sentence around how as part of week x of the F&C course you put together a tutorial that does x.

**Comment by iteles** [comment]
@rjmk Made some super simple suggestions, entirely optional :+1:  :blush: 

@sofer Are you still the [designated blog editor](https://github.com/foundersandcoders/playbook/issues/3)?

**Comment by rjmk** [comment]
@iteles Thanks, your suggestions are fantastic. I'll submit an updated version soon. :)

**Pull request merged**